returnn/tf/layers/basic.py

```diff
@@ -6097,16 +6091,61 @@ def __init__(self, condition, true_from, false_from, **kwargs):
     self.false_from = false_from
     if isinstance(condition, bool):
       if condition:
-        self.output = true_from.output.copy("%s_output" % self.name)
+        true_data = true_from.output if isinstance(true_from, LayerBase) else Data.from_tensor(tf.identity(true_from))
+        self.output = true_data.copy("%s_output" % self.name)
+        if self._get_source_allow_inf_in_output(true_from):
+            self.allow_inf_in_output = True
       else:
-        self.output = false_from.output.copy("%s_output" % self.name)
+        false_data = false_from.output if isinstance(false_from, LayerBase) else Data.from_tensor(
+          tf.identity(false_from))
+        self.output = false_data.copy("%s_output" % self.name)
+        if self._get_source_allow_inf_in_output(false_from):
+            self.allow_inf_in_output = True
```

Comment:
You have a lot of unnecessary duplicate code here. Don't do that if you can avoid that.

You could do:
```
src = true_from if condition else false_from
```
And then the same code.

Reply:
Also, indent is wrong.